egs/librispeech/ASR/conformer_ctc/utils/text_dataset.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python3
+
```

Comment:
Yes, that sounds good.

Reply:
> If you borrowed parts of this from ESPNet we should acknowledge that.




Currently, only the parameters are borrowed from ESPNet, (to make sure the model structure is identical to ESPNet and pretend we have a good trained model).

I think the code is not from ESPNet, because the LM is used in a different way with ESPNet.
In this pr, LM is used as a rescorer of a whole candidate sentence. 
While in ESPNet it is used in beam-search process during which P(Xt|Xo...Xt-1) is computed step by step. 

Anyway I will confirm this and add references in the code if needed.